OpenRA.Game/Player.cs

```diff
@@ -244,28 +249,40 @@ public bool IsAlliedWith(Player p)
 			return RelationshipWith(p) == PlayerRelationship.Ally;
 		}
 
-		public static Color PlayerRelationshipColor(Actor a)
+		/// <summary>Returns <see cref="color"/>, ignoring player relationship colors.</summary>
+		public static Color GetColor(Player p) => p.color;
+
+		public static void SetupRelationshipColors(Player[] players, Player observer)
 		{
-			var renderPlayer = a.World.RenderPlayer;
-			var player = renderPlayer ?? a.World.LocalPlayer;
-			if (player != null && !player.Spectating)
-			{
-				var effectiveOwner = a.EffectiveOwner;
-				var apparentOwner = a.Owner;
-				if (effectiveOwner != null && effectiveOwner.Disguised && !a.Owner.IsAlliedWith(renderPlayer))
-					apparentOwner = effectiveOwner.Owner;
+			foreach	(var p in players)
+				p.Color = PlayerRelationshipColor(p, observer);
+		}
 
-				if (apparentOwner == player)
-					return ChromeMetrics.Get<Color>("PlayerStanceColorSelf");
+		/// <summary>Returns <see cref="Color"/> of the <paramref name="actor"/>, taking <see cref="Actor.EffectiveOwner"/> into account.</summary>
+		public static Color ActorColor(Actor actor)
```

Comment:
1. `OwnerColor` would be clearer
2. This doesn't belong on `Player`. This and many of the other static methods here could be moved to a static `PlayerColorUtils` class instead.

Reply:
They are here as to not allow outside scripts to interfere. Moved OwnerColor out